src/store/modules/nutrition/actions.ts

```diff
@@ -11,17 +14,53 @@ export const actions: ActionTree<any, any> = {
                     day: payload,
                 });
             } else {
-                new NutritionAPI()
-                    .getMock(payload)
-                        .then((v: IDailyIntake) => {
-                            commit('cacheDailyIntake', {
-                                day: payload,
-                                intake: v,
-                            });
-                            commit('getDailyIntake', {
-                                day: payload,
-                            });
-                        });
+                try {
+                    const intake: IDailyIntake = await new NutritionAPI().getMock(payload) || {} as IDailyIntake;
+                    commit('cacheDailyIntake', {
+                        day: payload,
+                        intake,
+                    });
+                    commit('getDailyIntake', {
+                        day: payload,
+                    });
+                    return intake;
+                } catch (error) {
+                    throw (error);
+                }
             }
     },
+    async updateCurrentDailyIntake(
+        {state, commit}:
+            {state: NutritionState, commit: Commit},
+        payload:
+            {mealName: string, food: Food}) {
+                try {
```

Comment:
I'm not sure if action is the best place for catching errors from API. You could also catch them in the specific component while dispatching an action. That way you could separate showing error messages etc. from functionality specific to actions: calling API and commiting mutations.

Reply:
I will read about where to use try/catch clauses in application with vuex